node/network/availability-recovery/src/lib.rs

```diff
@@ -145,33 +145,37 @@ struct InteractionParams {
 	metrics: Metrics,
 }
 
-enum InteractionPhase {
-	RequestFromBackers(RequestFromBackersPhase),
-	RequestChunks(RequestChunksPhase),
+/// Source the availability data either by means
+/// of direct request response protocol to
+/// backers (a.k.a. fast-path), or recover from chunks.
+enum Sourcer {
+	RequestFromBackers(RequestFromBackersSourcer),
+	RequestChunks(RequestChunksSourcer),
 }
 
-/// A state of a single interaction reconstructing an available data.
-struct Interaction<S> {
+/// A stateful reconstruction of availability data in reference to
+/// a candidate hash.
+struct DataRecoveryTask<S> {
 	sender: S,
 
-	/// The parameters of the interaction.
-	params: InteractionParams,
+	/// The parameters of the recovery process.
+	params: DataRecoveryParams,
 
-	/// The phase of the interaction.
-	phase: InteractionPhase,
+	/// The sourcer to obtain the availbility data.
+	sourcer: Sourcer,
 }
 
-impl RequestFromBackersPhase {
+impl RequestFromBackersSourcer {
 	fn new(mut backers: Vec<ValidatorIndex>) -> Self {
 		backers.shuffle(&mut rand::thread_rng());
 
-		RequestFromBackersPhase { shuffled_backers: backers }
+		RequestFromBackersSourcer { shuffled_backers: backers }
 	}
 
 	// Run this phase to completion.
 	async fn run(
 		&mut self,
-		params: &InteractionParams,
+		params: &DataRecoveryParams,
```

Comment:
`Data` is so generic, that we could equally well just drop it. `DataRecovery` -> `Recovery`. Considering that this is the availability recovery subsystem, it should be clear anyways what is going to be recovered.